### ISSUE TITLE
Provide clear way to download notebook as PDF

### ISSUE THREAD

**Issue by yuvipanda** [berkeley-dsep-infra/datahub#1461]
The following download options are available for students:

<img width="403" alt="image" src="https://user-images.githubusercontent.com/30430/81810598-d871be80-9540-11ea-8f58-7673335d39b5.png">

This is way too many, and most of them are useless.

Instead, we should just have the following options:

- Notebook (.ipynb)
- Python (.py)
- PDF

Currently PDF requires LaTeX. I don't want us to use LaTeX. 

https://github.com/betatim/notebook-as-pdf is probably the best current way to generate PDFs from notebook files. We currently have [my version](https://github.com/yuvipanda/nbpdfexport) installed in a few hubs. We should standardize on this instead.

For other formats, we can offer the appropriate nbconvert command so folks can run and download it manually

**Comment by yuvipanda** [comment]
This will make @ericvd-ucb very happy

**Comment by yuvipanda** [comment]
Asked question in https://discourse.jupyter.org/t/remove-download-as-options-in-jupyter-notebook/4374

**Comment by ericvd-ucb** [comment]
Yes! - it will make me happy - and many confused newbies as well.
I guess I don't exactly get what the betatim workflow does , what "attached" means, and what the limitation of not opening in preview and have to have acrobat ( FWIW I have preview but not Acrobat) 

@ktakimoto  or  @chrispyles  Want to weigh in here?

**Comment by yuvipanda** [comment]
@ericvd-ucb i think @betatim's version should work exactly the same as mine, except it also lets you retrieve the original notebook.ipynb file from the PDF if you have the appropriate software. The PDF itself should open in preview or other PDF applications

**Comment by ktakimoto** [comment]
Seconding the happiness around cleaning up the pdf options!

@ericvd-ucb having the ipynb file attached to the pdf is intriguing as we think about uploading assignments to Gradescope. I think our initial idea was to ask student to upload the ipynb, then convert to pdf on Gradescope- with this version, could we do the reverse (students upload pdf, ipynb is extracted on Gradescope)?

re: preview question, I think our OSX users don't need to see the ipynb attachment as long as they can see the pdf (and don't get any weird error messages about hidden attachments- can't test this currently, as I'm on Windows).

Possibly tangential: Modules has started using @chrispyles [nb2pdf ](https://github.com/ucbds-infra/nb2pdf) for pdf conversion, since it's compatible with otter-grader and can remove non-graded portions of the notebook from the pdf. @yuvipanda @chrispyles Is it possible to get nb2pdf conversion added as a "Download as" menu option?

**Comment by chrispyles** [comment]
This is a great idea! Have had lots of issues with students downloading the wrong filetypes.

W/ regard to Gradescope, having students just upload a PDF with the notebook attached doesn't completely alleviate the issues. Either:
* students upload to a HW assignment which has no autograder, in which case there's no way to extract the notebook, or
* students upload to a programming assignment which doesn't support page selection, another problem we've been having.

So far, Otter's workaround has been to create a zip file containing just the notebook for students to download and upload to a Gradescope autograding assignment (this alleviates some downloading issues relating to file extension, encoding, etc.) and then having the GS autograder generate a PDF and sent a request to Gradescope's API to upload this as a submission for a HW assignment which supports automatic page selection.

Also in regards to @ktakimoto's comment about adding nb2pdf to the export menu, nb2pdf is based on @yuvipanda's nbpdfexport, which is currently the "PDF via Chrome" option. The main difference is that nb2pdf can filter notebooks so that only certain parts of the notebook are exported. It might be valuable to have e.g. a "Filtered PDF" option that uses nb2pdf or Otter Export (new in v1) but I think that having a notebook cell that generates these filtered PDFs is better so as not to clutter the menu and because filtered PDFs seem more centered around Gradescope's use case and can be done at different places in the pipeline.

**Comment by chrispyles** [comment]
Also @yuvipanda if we switch off of LaTeX is the plan to remove it from the Hub? I ask because Otter Export uses LaTeX to generate PDFs because currently, pandoc's HTML template doesn't honor CSS page breaks, meaning that in order to have a question-per-page structure you need to use LaTeX to export the PDF.

**Comment by yuvipanda** [comment]
With https://github.com/jupyter/nbconvert/pull/1273 this might finally come true!

I'll hopefully be able to respond to other comments shortly, now that this is possible.

**Comment by ericvd-ucb** [comment]
@yuvipanda seems like there are a few use cases here - let's keep talking about how this will play out for our students - this is definitely a key issue for modules users

**Comment by ericvd-ucb** [comment]
@yuvipanda @chrispyles Would love to check back in on plans / ideas here.  We have modules work and deployments that are lining up to deploy this semester and we always looking to simplify a download pdf workflow.

**Comment by ryanlovett** [comment]
I just read that the latest version of nbconvert has a `--to webpdf` option that uses Pyppeteer, https://nbconvert.readthedocs.io/en/latest/install.html#installing-chromium. So if this is sufficient then the other custom PDF converters can be removed. But it doesn't lessen the issue of having too many built-in PDF converters overall. :)
